File: leetcode/238/238.py
class Solution(object):
    def productExceptSelf(self, nums):
        p = 1
        leng = len(nums)
        output = []
        for i in range(0,leng):
            output.append(p)
            p = p * nums[i]
        p = 1
        print(output)
        for i in range(leng-1,-1,-1):
            output[i] = output[i] * p
            p = p * nums[i]
        return output
        

File: leetcode/problems/238/238.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int leng = nums.size();
        int l = 1;
        vector<int> newv;
        for(int i = 1; i < leng; i++){
            newv[i] = l;
            l *= nums[i];
        }
        int i = leng - 1;
        l = 1;
        while(i >= 0){
            newv[i] *= l;
            l *= nums[i];
            i -= 1;
        }
        return newv;
    }
};